File: Medium/Array/2161. Partition Array According to Given Pivot.py
# 2161. Partition Array According to Given Pivot
class Solution:
    def pivotArray(self, nums: List[int], pivot: int) -> List[int]:

        n = len(nums)
        p = 0
        ans = []

        for i in nums:
            if(i< pivot) : ans.append(i)
            elif(i == pivot) : p+=1
        
        for i in range(p):
            ans.append(pivot)
        
        for i in nums:
            if(i > pivot) : ans.append(i)
        
        return ans

File: Medium/Array/2161. Partition Array According to Given Pivot.cpp
// 2161. Partition Array According to Given Pivot
class Solution {
public:
    vector<int> pivotArray(vector<int>& nums, int pivot) {

        int n = nums.size();
        int count = 0;
        vector<int>left;
        vector<int>right;

        for(int i = 0; i<n;i++){
            if(nums[i] == pivot) count++;
            else if(nums[i] < pivot) left.push_back(nums[i]);
            else right.push_back(nums[i]);


        }
        int st = 0;

        for(auto i: left){
            nums[st] = i;
            st++;
        }
        for(int i = 0; i<count;i++){
            nums[st] = pivot;
            st++;
        }
        for(auto i: right){
            nums[st] = i;
            st++;
        }

        return nums;
        
    }
};
